### PR TITLE
Add pending migration checking middleware

### LINKED ISSUE #485: Add migration checking middleware

### PR DESCRIPTION
**Pull request by jerben**
Fix #485

### CODE REVIEW

File: sihl/src/web_migration.ml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    let%lwt migrations = pending_migrations () in
+    if List.length migrations > 0
+    then (
+      Logs.err (fun m ->
+          m "There are %d pending migrations" (List.length migrations));
+      Logs.info (fun m ->
+          m "Run 'sihl migrate' to apply the pending migrations");
+      if Core_configuration.is_production ()
+         (* The error handler shows internal server error 500 *)
+      then raise Exception
```
Comment (aronerben) [inline sihl/src/web_migration.ml:54]: Are pending migrations this severe? Maybe we shouldn't even allow starting the app if there are pending migrations?

File: sihl/src/web_migration.ml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+<head>
+  <meta charset="utf-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
+  <title>Pending migrations</title>
+  <style>
+    %s
+  </style>
+</head>
+
+<body class="antialiased">
```
Comment (aronerben) [inline sihl/src/web_migration.ml:23]: Where are all these class names coming from?

**Review by aronerben — COMMENTED** [review]
Please check

File: sihl/src/web_migration.ml (hunk 1 above)
Comment (jerben) [reply sihl/src/web_migration.ml:54]: That is actually a good question. Maybe the HTTP server shouldn't even start with pending migrations. The repositories assumes a different schema than there is and you have to expect runtime errors. What do you think?

I took this middleware idea from Rails, but with Sihl we have actual service lifecycles that we can use. :)

Comment (jerben) [reply sihl/src/web_migration.ml:54]: After thinking about it, I don't think we should restrict a Sihl app that much. Let's just print a warning in production for every request and fail in development using the middleware approach.

Comment (aronerben) [reply sihl/src/web_migration.ml]: Ok, I don't know what the best approach is, but isn't it dangerous allowing a prod app to start with missing migrations? I propose this: By default the app doesn't even start with missing migrations in either dev and prod. As a sihl user, you can disable that block with an env var "ALLOW_MISSING_MIGRATIONS" or similar. What do you think?

**Review by aronerben — CHANGES_REQUESTED** [review]
Please check comments

File: sihl/src/web_migration.ml (hunk 1 above)
Comment (jerben) [reply sihl/src/web_migration.ml]: It is more complex to define what it means for a Sihl app to start. You can run all sorts of commands involving all sorts of subsets of services. From what I read Rails allows you to start production with pending migrations, that is why we should allow it.

File: sihl/src/web_migration.ml (hunk 2 above)
Comment (jerben) [reply sihl/src/web_migration.ml:23]: From the imported hardcoded error style.

File: sihl/src/web_migration.ml (hunk 1 above)
Comment (aronerben) [reply sihl/src/web_migration.ml]: Could we not say that anything requiring the database cannot start?

File: sihl/src/web_migration.ml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  <div class="py-4 sm:py-12">
+    <div class="max-w-8xl mx-auto px-4 sm:px-6 lg:px-8">
```
Comment (aronerben) [inline sihl/src/web_migration.ml:25]: `max-w-8xl` does not seem to be a valid class, please remove

**Review by aronerben — CHANGES_REQUESTED** [review]
Please check

File: sihl/src/web_migration.ml (hunk 1 above)
Comment (jerben) [reply sihl/src/web_migration.ml]: We could say that the migration service doesn't start if there are pending migrations, but that seems to be too strict.

File: sihl/src/web_migration.ml (hunk 3 above)
Comment (jerben) [reply sihl/src/web_migration.ml]: I removed the custom template and just throw an exception.

Comment (aronerben) [reply sihl/src/web_migration.ml]: Haha that works :D

File: sihl/src/web_migration.ml (hunk 1 above)
Comment (aronerben) [reply sihl/src/web_migration.ml]: Ok, then let's leave it like you proposed. => Merging

**Review by aronerben — APPROVED** [review]

**Pull request merged**